superdesk/commands/data_updates.py

```diff
@@ -0,0 +1,259 @@
+
+
+class Downgrade(DataUpdateCommand):
+    '''
+
+    Runs the latest data update backward.
+    If `data_update_id` is given, runs all the data updates backward until the given one.
+
+    '''
+    def run(self, data_update_id=None, fake=False, dry=False):
+        super().run(data_update_id, fake, dry)
+        data_updates_files = self.data_updates_files
+        # check if there is something to downgrade
+        if not self.last_data_update:
+            print('No data update has been already applied')
+            return False
+        # drops updates which have not been already made (this is rollback mode)
+        data_updates_files = [update for update in data_updates_files if self.in_db(update)]
+
+        # if data_update_id is given, go until this update (drop previous updates)
+        if data_update_id:
+            if data_update_id not in data_updates_files:
+                print('Update %s can\'t be find. It may have been already downgraded' % (data_update_id))
+                return False
+            data_updates_files = data_updates_files[data_updates_files.index(data_update_id):]
+        # otherwise, just rollback one update
+        else:
+            print('No data update id has been provided. Dowgrading to previous version: %s'
+                  % self.last_data_update['name'])
+            data_updates_files = data_updates_files[len(data_updates_files) - 1:]
+        # apply data updates, in the opposite direction
+        for data_update_name in reversed(data_updates_files):
+            print('data update %s running backward...' % (data_update_name))
+            module_scope = self.compile_update_in_module(data_update_name)
+            # run the data update backward
+            if not fake:
+                module_scope.DataUpdate().apply('backwards')
+            if not dry:
+                # remove the applied data update from the database
+                self.data_updates_service.delete({'name': data_update_name})
+        if not data_updates_files:
+            print('No data update to apply.')
+
+
+class GenerateUpdate(superdesk.Command):
+    '''
+
+    Generate a file where to define a new data update
+
+    '''
+    option_list = [
+        superdesk.Option('--resource', '-r', dest='resource_name', required=True,
+                         help='Resource to update'),
+        superdesk.Option('--global', '-g', dest='global_update', required=False, action='store_true',
+                         help='This data update belongs to superdesk core'),
+    ]
+
+    def run(self, resource_name, global_update=False):
+        # create a data update file
+        timestamp = '%s%s' % (time.strftime('%Y%m%d-%H%M%S'), datetime.now().microsecond)
+        if global_update:
+            update_dir = MAIN_DATA_UPDATES_DIR
+        else:
+            update_dir = get_dirs(only_relative_folder=True)[0]
+        data_update_filename = os.path.join(update_dir, '{}_{}.py'.format(timestamp, resource_name))
```

Comment:
Theoretically a new update can have an older timestamp than the last one if they are generated on different systems (different time zones). I think an incremented version number is better in this case. If two PRs want to merge updates with the same version the devs can figure out the order of the updates and change their PRs. This version can also be used for retrieving applied updates in the proper order.